backend/cn/lib/sctypes.ml

```diff
@@ -171,7 +171,14 @@ let rec of_ctype (Ctype.Ctype (_, ct_)) =
     in
     let@ ret_ct = of_ctype ret_ct in
     return (Function ((ret_q, ret_ct), args, variadic))
-  | Ctype.FunctionNoParams _ -> fail
+
+  (* The meaning of a function with no parameters depends on the version of C,
+     but here we treat it as if it really has no parameter (i.e., like `void`)
+     which is what more recent versions of the standard do. *)
```

Comment:
(As below, this is ok as a workaround, but FunctionNoParams has to be treated specially, in the Cerberus frontend before CN.)